### ISSUE TITLE
Multi-module support in TFVC

### ISSUE THREAD

**Issue by ForNeVeR** [microsoft/azure-devops-intellij#376]
Currently, the TFVC part of the plugin doesn't support different modules/VCS roots. There're some calls to `project.getBaseDir()` and e.g. `CommandUtils.getPartialWorkspace(Project)` in our code base, which doesn't work well with multiple VCS roots (the VCS plugin in unable to perform the basic operations for such projects).

We'll need to change that part to address the issue, and make the plugin to rely on registered VCS roots instead of touching `project.getBaseDir()`.

This, in turn, may cause problems like #367.

**Comment by hoppermandr** [comment]
For issue #399 there isn't even multiple roots - only one.  However it is not the same as the project base directory.

**Comment by ForNeVeR** [comment]
The issue is still the same: currently, no roots other than the project root are supported. But they should be.

**Comment by hoppermandr** [comment]
Agreed. I just wanted to clarify that it doesn't have to be a huge complex project to be impacted by this bug.  Even projects with one content root - one module - one VCS root - can be affected.

In this sense, the bug name ("Multi-module (multi-VCS root) support in TFVC") is incorrect and misleading.

**Comment by ForNeVeR** [comment]
Thanks, I have updated the issue title and initial post body to accomodate that case better.

**Comment by hoppermandr** [comment]
Hi - I wanted to follow up on this issue.  This issue prevents TFS being usable at all in the IDE.  I am unable to check out, check in, compare, or perform any operations in the IDE because of this issue.

**Comment by erythana** [comment]
@ForNeVeR 
Sorry for pinging but just wondering, any news on this? 💯
